rules/predicates.py

```diff
@@ -144,11 +144,11 @@ def skip(self):
         """
         raise SkipPredicate()
 
-    def test(self, obj=NO_VALUE, target=NO_VALUE):
+    def test(self, obj=NO_VALUE, target=NO_VALUE, perm=NO_VALUE):
         """
         The canonical method to invoke predicates.
         """
-        args = tuple(arg for arg in (obj, target) if arg is not NO_VALUE)
+        args = tuple(arg for arg in (obj, target, perm) if arg is not NO_VALUE)
```

Comment:
I aggree. Just don't have the time or insight to do it now. Please reject this pull request.